portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,43 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.*;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  ArrayList<String> comments = new ArrayList<String>();
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>A work in progress...</h1>");
+    //response.setContentType("text/html;");
+    //response.getWriter().println("<h1>A work in progress...</h1>");
+
+    comments.add("This is just to practice JSON conversion");
+    comments.add("This would be the second message");
+    comments.add("It is ok that these are hard-coded");
+    comments.add("An extra would not hurt");
+    //Responding with json-converted ArrayList
+    String json = convertToJson(comments);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  private String convertToJson(ArrayList comments) {
```

Comment:
nit: ideally you should check that size of comments arraylist is at least 4 before doing the downstream processing.

Reply:
I can change this and recommit. Also, what is nit?

Reply:
nit is short for nitpicking, it refers to a small nitpicky suggestion which a code reviewer might make but not demand.